core/src/main/java/com/linecorp/armeria/server/healthcheck/HealthCheckServiceBuilder.java

```diff
@@ -256,6 +261,31 @@ public HealthCheckServiceBuilder updatable(HealthCheckUpdateHandler updateHandle
         return this;
     }
 
+    /**
+     * Adds a {@link HealthCheckUpdateListener} which is invoked when the healthiness of the {@link Server} is
+     * updated.
+     *
+     * @see #updatable(boolean)
+     * @see #updatable(HealthCheckUpdateHandler)
+     */
+    public HealthCheckServiceBuilder updateListener(HealthCheckUpdateListener updateListener) {
+        updateListenersBuilder.add(requireNonNull(updateListener, "updateListener"));
+        return this;
+    }
+
+    /**
+     * Disables setting healthy when the {@link Server} starts. The healthiness is updated using
+     * {@link HealthCheckUpdateHandler}. Please note that it's set unhealthy when the {@link Server} stops
+     * regardless.
+     *
+     * @see #updatable(boolean)
+     * @see #updatable(HealthCheckUpdateHandler)
+     */
+    public HealthCheckServiceBuilder disableServerListenerUpdate() {
```

Comment:
- `startUnhealthy()`?
- The healthiness is updated using -> This might be useful when you want to update the healthiness
  manually later via